kotlin-result/src/commonMain/kotlin/com/github/michaelbull/result/Result.kt

```diff
@@ -32,10 +32,10 @@ sealed class Result<out V, out E> {
 /**
  * Represents a successful [Result], containing a [value].
  */
-class Ok<out V>(val value: V) : Result<V, Nothing>() {
+public class Ok<out V>(public val value: V) : Result<V, Nothing>() {
 
-    override fun component1() = value
-    override fun component2() = null
+    override fun component1(): V = value
+    override fun component2(): Nothing? = null
```

Comment:
`Nothing?` 😑

Reply:
Seems right to me. If you try and destructure an `Ok<Int>` as such:

`val (good, bad) = Ok<Int>(5)`

`good` will be typed to `Int` and `bad` to `null`. Since you can't actually use `Nothing` as a variable, the only suitable approach is for the destructuring to return `null` (it can't return a `Nothing`).

Reply:
ye I'm just hung up on the semantics of "nullable nothing"

Reply:
I suppose if we made it throw an exception then you could make it return `Nothing` instead of `Nothing?`... but I don't expect people will want destructuring a `Result` to throw anything

Reply:
yep I agree having it throw would be bad - I think its more of a fallout from having the compiler allow all types be nullable even when you have something like `Nothing` that semantically expresses not being able to hold something 😅 .
Im sure the kotlin lang guys thought about this at some point during its creation, guess the pros just outweighed the cons for something so practical like nullability 🤷‍♂️